spec/spec_helper.rb

```diff
@@ -94,64 +104,48 @@ def current_password_reset_link
   inbox_unread.each do |email|
     next unless email.subject == 'Reset your password'
     msg = email.message.parts[0].body
-    url = msg.match(/(https:.+reset_password_token=[\w\-]+)/)[1]
+    url = msg.match(/(https:.+reset_password_token=[\w\-]+)/)
     if url
       email.read!
-      return url
+      return url[1]
     end
   end
   nil
 end
 
-def check_for_otp
-  otp = nil
+def check_inbox_for_email_value(label)
+  value = nil
   counter = 0
-  while (!otp) do
+  while (!value) do
     sleep 2
-    puts 'checking for OTP...'
-    otp = current_otp_from_email
+    puts "checking for #{label} ..."
+    value = yield
     counter += 1
     if counter >= 60
-      puts 'giving up OTP check ... timed out'
+      puts "giving up #{label} check ... timed out"
       break
     end
   end
-  fail 'Cannot find OTP' unless otp
-  otp
+  fail "cannot find #{label}" unless value
+  value
+end
+
+def check_for_otp
+  check_inbox_for_email_value('OTP') do
+    current_otp_from_email
+  end
```

Comment:
put this method above `check_inbox_for_email_value(label)` for top-down readability?